src/SSHDebugPS/AD7PortSupplier.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.VisualStudio.Debugger.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.SSHDebugPS
+{
+    [ComVisible(true)]
+    [Guid("1326FB0D-EA51-4F90-BEDF-5948588B0FE1")]
+    internal class AD7PortSupplier : IDebugPortSupplier2, IDebugPortSupplier3, IDebugPortSupplierDescription2
+    {
+        private const string Name = "SSH";
+        private readonly Guid _Id = new Guid("3FDDF14E-E758-4695-BE0C-7509920432C9");
+        public AD7PortSupplier()
+        {
+        }
+
+        public int AddPort(IDebugPortRequest2 request, out IDebugPort2 port)
+        {
+            string name;
+            HR.Check(request.GetPortName(out name));
+
+            port = new AD7Port(this, name, isInAddPort: true);
+            return HR.S_OK;
+        }
+
+        public int CanAddPort()
+        {
+            return HR.S_OK;
+        }
+
+        public int EnumPorts(out IEnumDebugPorts2 ppEnum)
+        {
+            liblinux.Persistence.ConnectionInfoStore store = new liblinux.Persistence.ConnectionInfoStore();
+            IDebugPort2[] ports = new IDebugPort2[store.Connections.Count];
+
+            for (int i = 0; i < store.Connections.Count; i++)
+            {
+                string hostName = ((liblinux.ConnectionInfo)store.Connections[i]).HostNameOrAddress;
```

Comment:
Lets add a TODO comment to investigate what happens if there are multiple connections to the same host (ex: I need to connect as root to debug some processes)